lib/ansible/modules/source_control/hg.py

```diff
@@ -231,13 +231,16 @@ def at_revision(self):
         There is no point in pulling from a potentially down/slow remote site
         if the desired changeset is already the current changeset.
         """
-        if self.revision is None or len(self.revision) < 7:
-            # Assume it's a rev number, tag, or branch
+        if self.revision is None:
             return False
+        # transform whatever was passed as revision into changeset ID
+        (rc, currentRevisionOut, err) = self._command(['--debug', 'id', '-i', '-r', self.revision, '-R', self.dest])
```

Comment:
The better command to use for this would be: `['log', '-T', '{node}', '-r', 'last(%s)' % self.revision, '-R', self.dest]`

Using `hg id --debug` means you might get unexpected output besides just the changeset hash. Same goes for line 240 below.